examples/c/ex_col_store.c

```diff
@@ -0,0 +1,585 @@
+        if (strcmp("AUS", country) == 0) {
+            printf("Removing %s\n", country);
+            error_check(cursor->remove(cursor));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+generate_data(WEATHER *w_array)
+{
+    int day;
+    int country;
+
+    srand((unsigned int)getpid());
+
+    for (int i = 0; i < NUM_ENTRIES; i++) {
+        WEATHER w;
+        day = rand() % 7;
+
+        switch (day) {
+        case 0:
+            strcpy(w.day, "MON");
+            break;
+        case 1:
+            strcpy(w.day, "TUE");
+            break;
+        case 2:
+            strcpy(w.day, "WED");
+            break;
+        case 3:
+            strcpy(w.day, "THU");
+            break;
+        case 4:
+            strcpy(w.day, "FRI");
+            break;
+        case 5:
+            strcpy(w.day, "SAT");
+            break;
+        case 6:
+            strcpy(w.day, "SUN");
+            break;
+        default:
+            assert(false);
+        }
+        w.hour = rand() % 2401;
+        w.temp = rand() % 51;
+        w.humidity = rand() % 101;
+        w.pressure = rand() % (1100 + 1 - 900) + 900;
+        w.wind = rand() % 201;
+        w.loc_lat = rand() % 181;
+        w.loc_long = rand() % 91;
+        country = rand() % 7;
+        switch (country) {
+        case 0:
+            strcpy(w.country, "AUS");
+            break;
+        case 1:
+            strcpy(w.country, "UK");
+            break;
+        case 2:
+            strcpy(w.country, "US");
+            break;
+        case 3:
+            strcpy(w.country, "NZ");
+            break;
+        case 4:
+            strcpy(w.country, "IND");
+            break;
+        case 5:
+            strcpy(w.country, "CHI");
+            break;
+        case 6:
+            strcpy(w.country, "RUS");
+            break;
+        default:
+            assert(false);
+        }
+
+        w_array[i] = w;
+    }
+}
+
+static bool
+find_min_temp(WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *result)
+{
+    WT_CURSOR *join_cursor, *start_time_cursor, *end_time_cursor;
+    int exact;
+    int ret = 0;
+    uint64_t recno;
+    uint16_t hour;
+    uint8_t temp;
+
+    /* Open cursors needed by the join. */
+    error_check(session->open_cursor(
+      session, "join:table:weathertable(hour,temp)", NULL, NULL, &join_cursor));
+    error_check(
+      session->open_cursor(session, "index:weathertable:hour", NULL, NULL, &start_time_cursor));
+    error_check(
+      session->open_cursor(session, "index:weathertable:hour", NULL, NULL, &end_time_cursor));
+
+    /*
+     * Select values WHERE (hour >= start AND hour <= end). Find the starting record closest to
+     * desired start time.
+     */
+    start_time_cursor->set_key(start_time_cursor, start_time);
+    error_check(start_time_cursor->search_near(start_time_cursor, &exact));
+    switch (exact) {
```

Comment:
This seems a bit weird. Can you just replace it with an `if` statement?

In my mind, `switch` is for dealing with multiple possible values.

Reply:
I had other cases in here before but forgot to clean this up, sorry. Thanks for the suggestion!